### PR TITLE
Learning Maps by implementing some CRUD over a dictionary

### PR DESCRIPTION
**Pull request by rjar2020**
What the heck is going on here?

- Trying maps
- Replacing `gorc` with plain, simple and recursive` go test ./...`
- Refactor pointer (wallet) benchmarks

That's all @JuanCalcagno416, apologies for the meaty PR.

### CODE REVIEW

File: maps/dictionary_test.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+	retrievedDefinition, err := dictionary.Search(word)
+	asserts.Nil(err)
+	asserts.Equal(definition, retrievedDefinition)
+}
+
+func BenchmarkDictionary_Add(b *testing.B) {
+	word := "test"
+	definition := "this is just a test"
+
+	//I'm not happy with this, as it's also adding up the exec time of randString()
```
Comment (rjar2020) [inline maps/dictionary_test.go:103]: I'm planning to fix this in a different PR, bear with me.

**Comment by rjar2020** [comment]
Additional interesting resource from the tuto for this chapter:

- https://dave.cheney.net/2016/04/07/constant-errors
- https://dave.cheney.net/2017/04/30/if-a-map-isnt-a-reference-variable-what-is-it
- https://blog.golang.org/maps

File: maps/dictionary.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	definition, ok := d[word]
+
+	if !ok {
+		return "", ErrNotFound
+	}
+
+	return definition, nil
+}
+
+//Search enables a client to add a new word's definition
```
Comment (JuanCalcagno416) [inline maps/dictionary.go]:
```suggestion
//Add enables a client to add a new word's definition
```

File: maps/dictionary.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	case nil:
+		d[word] = definition
+	default:
+		return err
+	}
+
+	return nil
+}
+
+//Update enables a client to delete a word's definition
```
Comment (JuanCalcagno416) [inline maps/dictionary.go]:
```suggestion
//Delete enables a client to delete a word's definition
```

Comment (rjar2020) [reply maps/dictionary.go]: El copy paste :-P

**Pull request merged**